WordPress/openverse-catalog
# Generate DAGs to recreate popularity calculations using a factory

## Fixes


Fast follow to #496

## Description


#496 combined the data refresh steps and the routine popularity data refresh tasks into a single DAG. But we also have a `recreate_popularity_calculations` DAG for each media type, which is _not_ run on any schedule and is only used manually to completely drop and recreate all the popularity relations and functions. We would use this when the SQL functions for calculating popularity change, for example.

This PR keeps those DAGs separate from the main data_refresh DAG to avoid complicating the standard workflow, but refactors them to be generated from a factory. As we add new media types, all we'll need to do is add a `DataRefreshType` to the config list in [data_refresh_types.py](https://github.com/WordPress/openverse-catalog/blob/main/openverse_catalog/dags/data_refresh/data_refresh_types.py#L53) and _all_ of the popularity/data refresh DAGs will be automatically generated.

I've also cleaned up a few things and added some docs, but this should not change any functionality.

<img width="1695" alt="Screen Shot 2022-05-13 at 2 59 28 PM" src="https://user-images.githubusercontent.com/63313398/168394781-c3443076-5c9d-4fb5-a7bd-c48f40e83b66.png">


## Testing Instructions

Run the `recreate_audio_popularity_calculation` and `recreate_image_popularity_calculation` DAGs and ensure they pass. 

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
